pkg/kubelet/kubelet.go

```diff
@@ -2103,9 +2110,6 @@ func (kl *Kubelet) HandlePodRemoves(pods []*v1.Pod) {
 			kl.handleMirrorPod(pod, start)
 			continue
 		}
-		if _, ok := kl.podManager.GetMirrorPodByPod(pod); ok {
-			kl.podKiller.MarkMirrorPodPendingTermination(pod)
-		}
```

Comment:
Ahh, I think I see what's happening here. This code never even gets executed because we'll hit the `continue` above before ever executing this.

My next question was "should this then go in `handleMirrorPod` instead?" but that code looks like it's in need of a refactor since 2015... https://github.com/kubernetes/kubernetes/issues/17251

Reply:
The problem I am pointing out is that this code was executed but the kl.deletePod below may return an error before executing podKiller.KillPod. And then we don't have a chance to delete the pending state from the podKiller.mirrorPodTerminationMap(a leak problem). What I fixed is ensuring to call the podKiller.KillPod.
https://github.com/kubernetes/kubernetes/blob/b6958b2b1bb2cdf9a7c1ef78f525a5305e455f54/pkg/kubelet/kubelet.go#L1738-L1772